### ISSUE TITLE
Inject database connection

### ISSUE THREAD

**Issue by dodtsair** [ThomWright/postgres-migrations#29]
Have you considered allowing the database connection to be injected?   I designing an application where  the database connection work is all collected into once place for use with all clients.  postgres-migrations would be one client.  This would give me a little more safety around managing database connection credentials (keys, passwords).  

In the current documentation the connection information is provided twice... why is that?

Either way, consider changing the example from:
const {createDb, migrate} = require("postgres-migrations")

```
createDb("database-name", {
  defaultDatabase: "postgres", // optional, default: "postgres"
  user: "postgres",
  password: "password",
  host: "localhost",
  port: 5432,
})
.then(() => {
  return migrate({
    database: "database-name",
    user: "postgres",
    password: "password",
    host: "localhost",
    port: 5432,
  }, "path/to/migration/files")
})
.then(() => {/* ... */})
.catch((err) => {
  console.log(err)
})
```

to the following, with the connection information removed, the connection created before hand, and then handed to postgres-migrations
```
const pg = require("pg")
const pgInitialized = require("pg-initializer").initialize(pg)
const {createDb, migrate} = require("postgres-migrations")

createDb("database-name", {
  connection: pgInitialized
})
.then(() => {
  return migrate({
    connection: pgInitialized
  }, "path/to/migration/files")
})
.then(() => {/* ... */})
.catch((err) => {
  console.log(err)
})
```

**Comment by ThomWright** [comment]
I like this. Keeping the connection outside of the library would be preferable. It can then be used however the application likes.

Would you be able to put together a PR?

To keep this from being a breaking change, I propose the function be able to take either the `{connection}` or the existing `DBConfig` types.

**Comment by ThomWright** [comment]
Solved by c246ad3 and ad81ed9
